### ISSUE TITLE
Donde poner cartas.py

### ISSUE THREAD

**Issue by FlorenciaOnetto** [IIC2233/Syllabus#367]
### Prerrequisitos
(Marcar colocando una X entre los corchetes los ítems que ya hiciste, así: "[X]")

* [ ] Leí las reglas del foro (https://github.com/IIC2233/syllabus/issues/1).
* [ ] Busqué en las *issues* si ya preguntaron mi duda y no encontré nada parecido (https://github.com/IIC2233/syllabus/issues/).
* [ ] Revisé el compilado de dudas de la tarea y no encontré una *issue* similar a la mía (https://github.com/IIC2233/syllabus/issues/326).
* [ ] Mi duda no se trata sobre una librería, *built-in* o mala práctica, ya que eso se pregunta en la *issue* creada para ello (https://github.com/IIC2233/syllabus/issues/325).
* [ ] Mi duda no se trata de un tema administrativo o personal, ya que en ese caso debo contactar a mi profe, al Jefe de Bienestar o al correo del curso (iic2233@ing.puc.cl).
* [ ] Utilizaré un título descriptivo y llenaré correctamente esta plantilla.
* [ ] De ser necesario, colocaré código simple que permita explicar mi problema o duda, ya que compartir código de esta evaluación puede ser considerado una falta a la ética.

### Duda

Hola!
No entiendo donde poner el archivo cartas.py, si en el Servidor o en el Cliente. En la página 5 del enunciado sale que tiene que estar en el servidor, pero no me hace sentido que esté ahí ya que cada jugador tiene su propia baraja de cartas, por lo que debería estar en la carpeta Cliente. Además, si estuviese en el Servidor, no entiendo como enviar un diccionario al Cliente.

**Comment by victoriaatria** [comment]
Hola!
Como mencionas, cartas.py debe efectivamente ir en el servidor. Recuerda que el servidor es el encargado de la interacción entre clientes y esto incluye asignarles sus respectivos mazos.  Los diccionarios se envían como cualquier otro tipo de dato, utilizando serialización y luego networking para enviar el mensaje. En los contenidos de las semanas 8 y 9 puedes encontrar ejemplos que te pueden ayudar a lograr esto!
Espero que te haya quedado más claro, quedo atenta por si tienes más dudas.

Saludos :)

**Comment by FlorenciaOnetto** [comment]
Lo que aun no me queda claro es como diferenciar el mensaje enviado
(diccionario) en el cliente. Lo que hacía yo con mensajes de strings era
usar una palabra clave al principio, pero ahora no se como diferenciar ese
mensaje ya que es un diccionario. No se si se entiende mi duda.

**Comment by victoriaatria** [comment]
Sí, se entiende tu duda. Hay muchas maneras de abordar eso! Podrías, por ejemplo, enviar siempre un mensaje inicial indicando el tipo de dato que enviarás (por supuesto este mensaje igual debe seguir todas las reglas de codificación y encriptación), o si prefieres enviar todo en un mismo mensaje, podrías agregar un string al principio de cada mensaje que indique el tipo de dato. Si te fijas, cuando serializamos los datos a un string, podemos concatenarlos juntos para crear un solo mensaje. De esta manera, podrías concatenar un string que originalmente era un diccionario, con otro string. Hay muchas otras maneras de abordar esto, para que le des otra vuelta.
Espero que te haya servido! Saludos
